subsys/bluetooth/controller/coex/coex_ticker.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2022 Dronetag
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <logging/log.h>
+#include <drivers/gpio.h>
+#include <sys/__assert.h>
+#include <bluetooth/hci.h>
+
+#include "controller/hal/ticker.h"
+#include "controller/ticker/ticker.h"
+#include "controller/include/ll.h"
+#include "controller/ll_sw/nordic/hal/nrf5/debug.h"
+
+LOG_MODULE_REGISTER(coex_ticker);
+
+#define COEX_RADIO_WORK_DELAY_US  50
+#define COEX_RADIO_WORK_RESCHEDULE_DELAY_US  200
+
+struct coex_ticker_config {
+	struct gpio_dt_spec grant_spec;
+	size_t grant_delay_us;
+};
+
+struct coex_ticker_data {
+	const struct device *dev;
+	struct gpio_callback grant_irq_cb;
+};
+
+static int time_slot_delay(uint32_t ticks_at_expire, uint32_t ticks_delay,
+				ticker_timeout_func callback, void *context)
+{
+	uint8_t instance_index;
+	uint8_t ticker_id;
+	int err;
+
+	ll_coex_ticker_id_get(&instance_index, &ticker_id);
+
+	/* start a secondary one-shot ticker after ticks_delay,
+	 * this will let any radio role to gracefully abort and release the
+	 * Radio h/w.
+	 */
+	err = ticker_start(instance_index, /* Radio instance ticker */
+				1, /* user id for link layer ULL_HIGH */
+				   /* (MAYFLY_CALL_ID_WORKER) */
+				ticker_id, /* ticker_id */
+				ticks_at_expire, /* current tick */
+				ticks_delay, /* one-shot delayed timeout */
+				0, /* periodic timeout  */
+				0, /* periodic remainder */
+				0, /* lazy, voluntary skips */
+				0,
```

Comment:
Why not use reservation in ticker? That is pass a ticks_slot value, this will ensure no other ticker (controller role) expires within this ticks_slot.

ticks_slot would be COEX_RADIO_WORK_RESCHEDULE_DELAY_US, right?


Thinking aloud here, the ticker can be a periodic ticker with  interval and ticks_slot set to the minimum duration coex is required (should I assume this is 200 us? COEX_RADIO_WORK_RESCHEDULE_DELAY_US?). If not, then have it a higher value.
Using ticks_slot ensure there is no redundant BLE radio use during these reserved time that is now owned by coex ticker.

Reply:
I will try to make the modifications based on your feedback.  
Since i don't have deep knowledge of the ticker system inside the controller.  I didn't event know there is this option.

Hopefully i will get to it this week.

Reply:
The PR is good as-is as a first spin, and I assume that it meets your requirements. You can address the minor change requests  and we should be able to merge this with others reviewers approvals.

In a follow up PR, you can do the improvements that will reduce CPU/Radio use (reduce overall aborts).